### PR TITLE
Restructure ZVM, HLZASM and GPULib

### PR DESCRIPTION
**Pull request by its-suun**
I maybe forgot the pull request and then I broke something with git. I renamed the files like @AbigailBuccaneer said in #1122. Tested CPU and GPU.

### CODE REVIEW

**Comment by Divran** [comment]
So you renamed the file of gpulib to screenlib without renaming the library itself? Wouldn't that make it more confusing, not less? I mean if you're looking in some other code that uses gpulib and you see the code looks like this `GPULib.Render` or whatever, then you'd probably go look for a file named gpulib. @AbigailBuccaneer what do you think

**Comment by its-suun** [comment]
Renaming GPULib to ScreenLib would probably break lots of code of addons which use wiremod's gpulib

**Comment by Divran** [comment]
Yeah but you could add this to the bottom of the file `GPULib = ScreenLib`

**Comment by its-suun** [comment]
Well if I would change it to ScreenLib then i would at least put it in Wirelib, like `WireLib.Screen`. Then we would have one global less

File: lua/autorun/wire_load.lua [hunk 1]
```diff
@@ -63,21 +63,21 @@ if SERVER then
...
-	AddCSLuaFile("wire/client/hlzasm/hc_tokenizer.lua")
+	AddCSLuaFile("wire/zyelios/hlzasm/hc_compiler.lua")
+	AddCSLuaFile("wire/zyelios/hlzasm/hc_opcodes.lua")
+	AddCSLuaFile("wire/zyelios/hlzasm/hc_expression.lua")
+	AddCSLuaFile("wire/zyelios/hlzasm/hc_preprocess.lua")
+	AddCSLuaFile("wire/zyelios/hlzasm/hc_syntax.lua")
+	AddCSLuaFile("wire/zyelios/hlzasm/hc_codetree.lua")
+	AddCSLuaFile("wire/zyelios/hlzasm/hc_optimize.lua")
+	AddCSLuaFile("wire/zyelios/hlzasm/hc_output.lua")
+	AddCSLuaFile("wire/zyelios/hlzasm/hc_tokenizer.lua")
```
Comment (AbigailBuccaneer) [inline lua/autorun/wire_load.lua]: This change doesn't make much sense - keeping client-only files in `wire/client/` is preferable.

("Zyelios" is also quite a confusing term, but I can see the rationale behind wanting to group `hlzasm` and `zvm` under one directory.)

File: lua/entities/gmod_wire_cpu.lua [hunk 2]
```diff
@@ -82,11 +82,11 @@ function ENT:Initialize()
...
-			else return 0 
+			else return 0
 			end
 		end
 	end
-	
+
 	local oldReset = self.VM.Reset
 	self.VM.Reset = function(...)
 		if self.Clk and self.VMStopped then
```
Comment (AbigailBuccaneer) [inline lua/entities/gmod_wire_cpu.lua]: This file is nothing but whitespace changes, it shouldn't be in the pull request.

File: lua/wire/zyelios/hlzasm/hc_compiler.lua [hunk 3]
```diff
@@ -26,14 +26,14 @@ HCOMP = {}
...
 -- Files required by the compiler
-include("wire/client/hlzasm/hc_opcodes.lua")
-include("wire/client/hlzasm/hc_expression.lua")
-include("wire/client/hlzasm/hc_preprocess.lua")
-include("wire/client/hlzasm/hc_syntax.lua")
-include("wire/client/hlzasm/hc_codetree.lua")
-include("wire/client/hlzasm/hc_optimize.lua")
-include("wire/client/hlzasm/hc_output.lua")
-include("wire/client/hlzasm/hc_tokenizer.lua")
+include("wire/zyelios//hlzasm/hc_opcodes.lua")
```
Comment (AbigailBuccaneer) [inline lua/wire/zyelios/hlzasm/hc_compiler.lua]: `//` → `/`

File: lua/autorun/wire_load.lua (hunk 1 above)
Comment (AbigailBuccaneer) [inline lua/autorun/wire_load.lua]: That being said, you did implement exactly what I suggested in the previous pull request - sorry for being so inconsistent :stuck_out_tongue_winking_eye:

**Comment by AbigailBuccaneer** [comment]
Regardless of whether `GPULib` gets renamed, I think that's a change that should happen in another pull request - remember it's easier for PRs to get accepted if they have less stuff for devs to argue about! :wink:

**Comment by AbigailBuccaneer** [comment]
Can we place things in `lua/includes/modules/` and then load them with `require` like Garry's modules? I'm pretty sure `include` parses and runs the file again even if it's already been loaded once, whereas `require` should sidestep that. I'll look into this when I have free time.

**Comment by Divran** [comment]
>Regardless of whether GPULib gets renamed, I think that's a change that should happen in another pull request - remember it's easier for PRs to get accepted if they have less stuff for devs to argue about! :wink:

Yeah, don't change the name of the gpulib library in this pr so we can take some time to agree on what to do there

>Can we place things in lua/includes/modules/ and then load them with require like Garry's modules? I'm pretty sure include parses and runs the file again even if it's already been loaded once, whereas require should sidestep that. I'll look into this when I have free time.

This doesn't matter though since we only ever include the files once anyway

**Comment by its-suun** [comment]
> This doesn't matter though since we only ever include the files once anyway  

ZVM gets included each time a new VM gets created (cpu spawned). [#726](https://github.com/wiremod/wire/blob/master/lua/wire/cpulib.lua#L726)

**Comment by Divran** [comment]
Oh well then

**Comment by TomyLobo** [comment]
maybe *that* ahould be fixed, too, then?

**Comment by its-suun** [comment]
Well, how should the directory structure be now?

**Pull request closed without merging**